pkg/build/linux_linux.go

```diff
@@ -65,6 +68,23 @@ func embedLinuxKernel(params Params, kernelPath string) error {
 	return osutil.CopyFile(imageFile, filepath.Join(params.OutputDir, "image"))
 }
 
+func mountRootPartition(loopFile, mountDir string) error {
+	etcFolder := filepath.Join(mountDir, "etc")
+	if osutil.IsExist(etcFolder) {
+		return nil
+	}
+	for idx := 2; ; idx++ {
+		err := tryMount(fmt.Sprintf("%sp%d", loopFile, idx), mountDir)
+		if err != nil {
+			return err
+		}
+		if osutil.IsExist(etcFolder) {
```

Comment:
If you move this to the beginning of the loop, you can remove the check in the beginning of the function.